packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -6536,7 +6538,7 @@ abstract class RenderObjectElement extends Element {
             ErrorSummary('Incorrect use of ParentDataWidget.'),
             ...parentDataWidget._debugDescribeIncorrectParentDataType(
               parentData: renderObject.parentData,
-              parentDataCreator: _ancestorRenderObjectElement!.widget as RenderObjectWidget,
+              parentDataCreator: _ancestorRenderObjectElement?.widget as RenderObjectWidget?,
```

Comment:
The `!` was originally hiding the real failure. Since parentDataCreator is allowed to be null, I changed this to `?` to get the real failure message right away should we run into a similar problem elsewhere.